clippy_lints/src/arithmetic.rs

```diff
@@ -82,7 +82,8 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Arithmetic {
                     _ => (),
                 }
                 let (l_ty, r_ty) = (cx.tables.expr_ty(l), cx.tables.expr_ty(r));
-                if l_ty.is_integral() && r_ty.is_integral() {
+                // warn on integers as well as references to integers
+                if l_ty.is_integral() || l_ty.is_any_ptr() && r_ty.is_integral() || r_ty.is_any_ptr() {
```

Comment:
Doesn't this now lint also for `&f32 + &f32`. Reading the [`is_any_ptr`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/struct.TyS.html#method.is_any_ptr) documentation:

> Tests if this is any kind of primitive pointer type (reference, raw pointer, fn pointer).

I think it does.

Reply:
Mmh good point. I believe I have a more elegant solution now.